app/models/lib/base_event.rb

```diff
@@ -20,12 +17,25 @@ def apply(aggregate)
     raise NotImplementedError
   end
 
+  # Aggregate name the event will belong to
+  # Should be a symbol. Example: `:user`
+  def self.aggregate_name
+    raise NotImplementedError, "Events must belong to an aggregate"
+  end
+
+  # Replays the event on any given aggregate
+  # @param [Object] aggregate
+  def replay(aggregate)
+    event_class = self.metadata["klass"].constantize
+    event_class.new(self.data).apply(aggregate)
+  end
```

Comment:
Also, if we don't want any `ActiveSupport` dependencies, this could be rewritten as `Object.const_get(self.metadata["klass"])`.

Reply:
I wonder if maybe it would be worth de-coupling the "calculator" (apply) code from what is effectively the persistence layer. Maybe we can get rid of the inheritance, so we won't need this method on the base class in the first place? (and get it out of the persistence layer).

I might have a play with those ideas in another PR. Either way what you've done is a big improvement!